### PR TITLE
[query] Allow GET + POST for read APIs

### PR DESCRIPTION
**Pull request by schallert**
**What this PR does / why we need it**: Allow POST requests in coordinator read APIs. Supports common use cases such as Grafana datasources that use POST.

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
Allow query requests to be GET or POST
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/query/api/v1/handler/prometheus/native/read_instantaneous.go [hunk 1]
```diff
@@ -40,9 +40,15 @@ const (
 	// handler, this matches the  default URL for the query endpoint
 	// found on a Prometheus server
 	PromReadInstantURL = handler.RoutePrefixV1 + "/query"
+)
 
-	// PromReadInstantHTTPMethod is the HTTP method used with this resource.
-	PromReadInstantHTTPMethod = http.MethodGet
+var (
+	// PromReadInstantHTTPMethods is the valid HTTP methods used with this
```
Comment (arnikola) [inline src/query/api/v1/handler/prometheus/native/read_instantaneous.go]: meganit: `PromReadInstantHTTPMethods are the valid HTTP methods...`

File: src/query/api/v1/handler/prometheus/native/read.go [hunk 2]
```diff
@@ -49,14 +49,18 @@ const (
...
-	PromReadHTTPMethod = http.MethodGet
-
 	// TODO: Move to config
 	initialBlockAlloc = 10
 )
 
 var (
+	// PromReadHTTPMethods is the valid HTTP methods used with this
+	// resource.
+	PromReadHTTPMethods = []string{
```
Comment (arnikola) [inline src/query/api/v1/handler/prometheus/native/read.go]: meganit: `PromReadHTTPMethods are the valid HTTP methods...`

File: src/query/api/v1/handler/prometheus/native/common_test.go [hunk 3]
```diff
@@ -86,6 +87,16 @@ func TestParamParsing(t *testing.T) {
 	require.Equal(t, promQuery, r.Query)
 }
 
+func TestParamParsing_POST(t *testing.T) {
+	params := defaultParams().Encode()
+	req := httptest.NewRequest("POST", PromReadURL, strings.NewReader(params))
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+	r, err := testParseParams(req)
+	require.Nil(t, err, "unable to parse request")
```
Comment (arnikola) [inline src/query/api/v1/handler/prometheus/native/common_test.go]: meganit: `require.NoError` gives slightly more context

**Comment by robskillington** [comment]
Going to merge this in, @arnikola will take up the nits as followups

**Pull request merged**